inst/include/dust/dust.hpp

```diff
@@ -863,17 +837,22 @@ class Dust {
   }
 
   void update_device_index() {
-    if (device_id_ < 0) {
+    if (!device_config_.enabled_) {
       return;
     }
-    device_state_.set_device_index(index_, particles_.size(), n_state_full());
+    const size_t n_particles = particles_.size();
+    device_state_.set_device_index(index_, n_particles, n_state_full());
 
     select_needed_ = true;
     if (!std::is_sorted(index_.cbegin(), index_.cend())) {
       select_scatter_ = true;
     } else {
       select_scatter_ = false;
     }
+
+    // TODO: get this 64 from the original configuration, if possible.
+    cuda_pars_.index_scatter =
+      dust::cuda::launch_control_simple(64, n_particles * n_state());
```

Comment:
If we had the defaults as consts we could use the variable to replace 64 here too